### ISSUE TITLE
Add support for `\dot (U+0307)` for numbers

### ISSUE THREAD

**Issue by hyrodium** [cormullion/juliamono#196]
Thank you for creating JuliaMono!

I would like to use `123.456̇78̇` format to represent a repeating decimal in my package [RepeatingDecimalNotations.jl](https://github.com/hyrodium/RepeatingDecimalNotations.jl), but `\dot (U+0307)` on numbers are not rendered correctly.

```julia
julia> # ȧ (`a\dot`), nice!

julia> # 0̇ (`0\dot`), the place of dot is not correct.

julia> # 123.456̇78̇ (`123.456\dot78\dot`)
```

![image](https://github.com/cormullion/juliamono/assets/7488140/08df857b-5349-47d2-80e3-abc8aed23f22)

Is it possible to resolve this issue with JuliaMono?

**Comment by cormullion** [comment]
Hi! The numerals don't normally have the top/bottom anchor positions set, but I've just added them to my working master. Here is VS-Code now:

<img width="727" alt="Screenshot 2023-12-23 at 17 19 44" src="https://github.com/cormullion/juliamono/assets/52289/662f4ab8-1670-426c-b391-a7b6b4d77475">

This feature will be in the next release of JuliaMono (hopefully ready by the end of December).

However...

The problem with this feature is that it depends on both the font **and** the application/terminal **and** the platform/OS. Some terminals won't display these correctly. For example, in the current release of Kitty on MacOS or Alacritty this glyph won't appear above the preceding glyph - even if the font supports the anchor points. Whereas Terminal, iTerm, Wezterm will. In fact, MacOS Terminal already positions these dots over the preceding glyph correctly with the current release of JuliaMono. In your testing you will no doubt encounter many combinations of font/application that work or don't work!

By the way, your package icon could do with a dark-mode compatible version! :)

**Comment by hyrodium** [comment]
>Hi! The numerals don't normally have the top/bottom anchor positions set, but I've just added them to my working master. Here is VS-Code now:
>This feature will be in the next release of JuliaMono (hopefully ready by the end of December).

Thank you for the quick response!
I'm really looking for it!

>The problem with this feature is that it depends on both the font and the application/terminal and the platform/OS. Some terminals won't display these correctly. For example, in the current release of Kitty on MacOS or Alacritty this glyph won't appear above the preceding glyph - even if the font supports the anchor points.

I see. Thank you for the detailed explanation! I don't fully understand how fonts work, and I have an additional question. Does this mean that a case like `a\dot` is supported while `0\dot` is not?

>By the way, your package icon could do with a dark-mode compatible version! :)

Thanks! I've updated the logo! :smile:

**Comment by cormullion** [comment]
>Does this mean that a case like a\dot is supported while 0\dot is not?

The situation is similar (depends on font/application): but, it's more likely that `a` + `mark` works than `0` + `mark` does. This is because French/German etc. languages require accents and diacritics on letters but not on numerals. For example, in VS-Code editors and terminals, you can expect FiraCode to show `àb̈` and similar, but probably not `1̀2̈`. (It also works in this Github issue too, for me...) 

I don't really know why things don't work as well as they should. Everything about fonts is less than 100% reliable in my experience! :)

**Comment by hyrodium** [comment]
Thank you for the fix!
I have confirmed that `\dot` was fixed in the v0.053 release.

![image](https://github.com/cormullion/juliamono/assets/7488140/7e6098c9-f1f1-4ccf-9327-a678b564c015)
